### PR TITLE
Rearrange Code to fix Circular Imports

### LINKED ISSUE #3616: Cannot Migrate Application Locally
### Bug Description

There has been some rearranging of the directory structure lately and now you can no longer run or generate migrations locally.  

This method works: https://github.com/ethyca/fides/blob/main/src/fides/api/db/database.py#LL26C2-L26C2 however which is why our tests are passing:
> The migration commands within the application run via Alembic's Python API and the Config has already been loaded at that time so I'm assuming that's why it works there and doesn't work via the CLI?

#### Steps to Reproduce

- nox -s dev -- shell.
- cd src/fides/api
- alembic revision -m "test_migration" --autogenerate

```
Traceback (most recent call last):
  File "/opt/fides/bin/alembic", line 8, in <module>
    sys.exit(main())
  File "/opt/fides/lib/python3.10/site-packages/alembic/config.py", line 590, in main
    CommandLine(prog=prog).main(argv=argv)
  File "/opt/fides/lib/python3.10/site-packages/alembic/config.py", line 584, in main
    self.run_cmd(cfg, options)
  File "/opt/fides/lib/python3.10/site-packages/alembic/config.py", line 561, in run_cmd
    fn(
  File "/opt/fides/lib/python3.10/site-packages/alembic/command.py", line 322, in upgrade
    script.run_env()
  File "/opt/fides/lib/python3.10/site-packages/alembic/script/base.py", line 569, in run_env
    util.load_python_file(self.dir, "env.py")
  File "/opt/fides/lib/python3.10/site-packages/alembic/util/pyfiles.py", line 94, in load_python_file
    module = load_module_py(module_id, path)
  File "/opt/fides/lib/python3.10/site-packages/alembic/util/pyfiles.py", line 110, in load_module_py
    spec.loader.exec_module(module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/fides/src/fides/api/migrations/env.py", line 6, in <module>
    from fides.api.util.logger import setup as setup_fidesapi_logger
  File "/fides/src/fides/api/util/logger.py", line 15, in <module>
    from fides.core.config import CONFIG
  File "/fides/src/fides/core/config/__init__.py", line 28, in <module>
    from .security_settings import SecuritySettings
  File "/fides/src/fides/core/config/security_settings.py", line 10, in <module>
    from fides.api.api.v1.scope_registry import SCOPE_REGISTRY
  File "/fides/src/fides/api/api/v1/__init__.py", line 8, in <module>
    from .endpoints.admin import ADMIN_ROUTER
  File "/fides/src/fides/api/api/v1/endpoints/admin.py", line 8, in <module>
    from fides.api.db.database import configure_db, reset_db
  File "/fides/src/fides/api/db/database.py", line 15, in <module>
    from fides.api.db.base import Base  # type: ignore[attr-defined]
  File "/fides/src/fides/api/db/base.py", line 5, in <module>
    from fides.api.models.application_config import ApplicationConfig
  File "/fides/src/fides/api/models/application_config.py", line 18, in <module>
    from fides.core.config import CONFIG, FidesConfig
ImportError: cannot import name 'CONFIG' from partially initialized module 'fides.core.config' (most likely due to a circular import) (/fides/src/fides/core/config/__init__.py)
```

### Expected behavior

We need to be able to generate migrations and do `alembic upgrade head` and `alembic downgrade -1` locally!

### Screenshots

_If applicable, add screenshots to help explain your problem._

### Environment

* Version:
* OS:
* Python Version:
* Docker Version:

### Additional context

_Add any other context about the problem here._

### PR DESCRIPTION
**Pull request by ThomasLaPiana**
Closes #3616 

### Code Changes

* [x] move `scope_registry` to `fides.common.api`
* [x] move `config` up one level to be a top-level module (`fides.config`)
* [x] remove `echo_green` and `echo_red` from `fides.core.utils`
* [x] move the 'alembic' files into their own directory within `fides.api`

### Steps to Confirm

* [ ] _list any manual steps for reviewers to confirm the changes_

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### Description Of Changes

With increasing frequency we've been running into circular import issues, so this PR does more code reorganization (I know I promised last time would be the last time....) to try to fix current issues as well as potential future issues.

The key to preventing circular imports is to isolate files that require dependencies across the application.

### CODE REVIEW

**Comment by ThomasLaPiana** [comment]
@adamsachs I _think_ I got the circular dependencies all fixed, but now I'm hitting this error when running the migration command

```
 from cryptography.hazmat.primitives.ciphers.aead import AESGCM
ModuleNotFoundError: No module named 'cryptography.hazmat'
```

**Comment by pattisdr** [comment]
Thanks for your work so far @ThomasLaPiana.  Just flagging this will cause corresponding fidesplus changes to be made too since we've changed the locations of some pieces that are imported there.

**Comment by adamsachs** [comment]
hmm, seems like our own `src/fides/api/cryptography/__init__.py` module is what the `cryptography` import is resolving to within the dev docker image, rather than `cryptography` resolving to the 3rd party package. trying to track down what could have changed recently in our local setup to make that be the case...

File: src/fides/api/alembic/migrations/versions/77c4a1e836d3_test_migration.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    pass
+    # ### end Alembic commands ###
+
+
+def downgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    pass
+    # ### end Alembic commands ###
```
[the remark this answers is not in the document]
Comment (adamsachs) [reply src/fides/api/alembic/migrations/versions/77c4a1e836d3_test_migration.py:28]: 🤦 clearly was too hasty in my `git` commands 😆 

thank you for catching!

[the remark this answers is not in the document]
Comment (adamsachs) [reply src/fides/api/alembic/migrations/versions/77c4a1e836d3_test_migration.py]: fixed in [cbe8dc7](https://github.com/ethyca/fides/pull/3621/commits/cbe8dc76fd1f5da5c89cf9bb179c3a2dd3c17012)

**Comment by ThomasLaPiana** [comment]
> hmm, seems like our own `src/fides/api/cryptography/__init__.py` module is what the `cryptography` import is resolving to when at `/fides/src/fides/api` within the docker image, rather than `cryptography` resolving to the 3rd party package/library we expect. i think this is just a product of code reorg bringing up a naming conflict, since before we didn't have a `cryptography` in the same dir as the `alembic.ini` where we run our alembic commands from.
> 
> could we just move the `alembic.ini` (and perhaps the `/migrations` subdir it relies upon) into a dedicated `alembic` or something similar subdir to prevent any namespace conflicts?

This makes a ton of sense, might as well!

from `The Zen of Python`:
> Flat is better than nested

**Comment by ThomasLaPiana** [comment]
@adamsachs @pattisdr good to go for another review! Looks good to me :)

**Review by adamsachs — APPROVED** [review]
looking good! great collab getting this worked out quickly 💪

**Comment by pattisdr** [comment]
hooray @ThomasLaPiana and @adamsachs thanks again

**Pull request merged**